### DISCUSSION TITLE
Chording non-English characters

### DISCUSSION THREAD

**Discussion by Nooby-001** [psoukie/zipchord#69] [Q&A] [answered]
Quicky: How can I set a chord to type ñ (ALT+0241)? 
{ALT}+0241 doesn't work, neither does "!"

**Comment by psoukie** [comment]
The dictionary file needs to be saved as UTF-8 with BOM. (I'm afraid the file encoding of the default English dictionary is without BOM.) To output the character "ñ" you can then define any chord such as:
```
n;   ñ
```
(with tab in between)
Or even as "` n   ñ`" (Space+N).

If you use smart spaces before / after chords and want to output the character without spaces, use the prefix/suffix notation:

```
chord   ~ñ~
```

(Unrelated: To use extended characters in chords -- on keyboard layouts which have them as unmodified keys -- you would create a custom keyboard and language setting.)

**Comment by Nooby-001** [comment]
Nothing worked but I found the solution by chance! 
use {U+00F1} Just google Unicode symbols or go here: https://unicode-table.com/en.

**Reply by psoukie** [reply]
This is a nice workaround! You should be able to use Unicode characters directly though.  See my comment below.

**Comment by psoukie** [comment, marked as answer]
I changed the encoding of dictionaries to UTF-8 with BOM in version 1.9.5 and 2.0-Alpha 4. This should solve the problem. (But existing dictionaries that do not work correctly with Unicode / extended characters need to be resaved with BOM -- On Windows 10 and later, in Notepad, after File > Save As, there should be an option in the "Encoding" dropdown for "UTF-8 with BOM.")

That way, you should be able to have any Unicode character directly in the dictionary without codes. With the attached test file,  I can have "ñ" directly (set to a chord of `Space`+`N`) 

[chords-test.txt](https://github.com/psoukie/zipchord/files/10334600/chords-test.txt)

**Reply by Nooby-001** [reply]
Thank you!

**Comment by AsmanovLev** [comment]
little request - it would be great if program could read unicode characters as keystokes (example wont work):
```
првт	привет
ку	привет
длшк	делишки
дла	дела
пркрсн	прекрасно
тличн	отлично
ужсн	ужасно
```
(example will work):
```
qq	привет
```

**Reply by psoukie** [reply]
Sorry for the delay in replying. Yes, this is possible. Currently, it requires you to create a new keyboard / language setting in ZipChord settings.

See the documentation on how to do that here: [Keyboard and language settings](https://github.com/psoukie/zipchord/wiki/Keyboard-and-language-settings)

(I have an automated detection of the keyboard layout on the roadmap but I could not spend much time on the development this year.)

Let me know if you run into any issues with it.
